atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -205,5 +130,58 @@ private void checkMatchingId(TimestampBoundStoreEntry entryInDb) {
         return Optional.of(col);
     }
 
+    @Override
+    public void throwGettingMultipleRunningTimestampServiceError(UUID id, TimestampBoundStoreEntry entryInDb) {
+        String message = "Error getting the timestamp limit from the DB: the timestamp service ID {} in the DB"
+                + " does not match this service's ID: {}. This may indicate that another timestamp service is"
+                + " running against this cassandra keyspace.";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                entryInDb.getIdAsString(), id.toString()}).getMessage();
+        logMessage(formattedMessage);
+        throw new MultipleRunningTimestampServiceError(formattedMessage);
+    }
+
+    @Override
+    protected void throwStoringMultipleRunningTimestampServiceError(long currentLimit, UUID id,
+            TimestampBoundStoreEntry entryInDb, TimestampBoundStoreEntry desiredNewEntry) {
+        String message = "Unable to CAS from {} to {}. Timestamp limit changed underneath us or another timestamp"
+                + " service ID detected. Limit in memory: {}, this service's ID: {}. Limit stored in DB: {},"
+                + " the ID stored in DB: {}. This may indicate that another timestamp service is running against"
+                + " this cassandra keyspace. This is likely caused by multiple copies of a service running without"
+                + " a configured set of leaders or a CLI being run with an embedded timestamp service against"
+                + " an already running service.";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                Long.toString(currentLimit),
+                desiredNewEntry.getTimestampAsString(),
+                Long.toString(currentLimit),
+                id.toString(),
+                entryInDb.getTimestampAsString(),
+                entryInDb.getIdAsString()})
+                .getMessage();
+        logMessage(formattedMessage);
+        throw new MultipleRunningTimestampServiceError(formattedMessage);
+    }
 
+    @Override
+    protected void throwNewTimestampTooSmallException(long currentLimit, TimestampBoundStoreEntry entryInDb) {
+        String message = "Cannot set cached timestamp bound value from {} to {}. The bounds must be increasing!";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                Long.toString(currentLimit), entryInDb.getTimestampAsString()}).getMessage();
+        logMessage(formattedMessage);
+        throw new IllegalArgumentException(formattedMessage);
+    }
```

Comment:
These 2 methods (and possibly the next 2) feel like they should belong in the abstract class - we already have the general `TimestampBoundStoreEntry` here.